src/libs/actions/Report.js

```diff
@@ -727,15 +733,16 @@ function fetchAllReports(
                 return;
             }
 
-            // The string cast here is necessary as Get rvl='chatList' may return an int
-            reportIDs = String(response.chatList).split(',');
```

Comment:
This produced a bug when `response.chatList` is empty `[]`. It would assign `['']` to report IDs, and the check ` if (reportIDs.length)` would be positive even though there are no reports

You can try this to see the outcome is `['']`

```js
console.log(String([]).split(',')) // prints ['']
```

---

Casting to string is no longer necessary as following usages of `reportIDs` work with both strings and numbers.
But if you need to have string IDs you can instead use map

```js
reportIDs = response.chatList.map(n => String(n)) // or n.toString()
```

Reply:
To clarify, the string cast is necessary because the API can return a `Number` (could be fixed eventually, but we are working around it for now).

Reply:
Yes but the only usages have no problems with numbers:

`fetchChatReportsByIDs(reportIDs);` would convert everything to string anyway since internally it makes this `chatList.join(',')`

`fetchActions(reportID);` seems to work OK with number or string

And merging to Onyx would again use `String`

```js
const currentReportID = firstReportID ? String(firstReportID) : '';
```

Reply:
I'm still a little confused sorry. When we get the `response.chatList` from `API.Get` it can be a number or a string and must be turning into an `Array`. `fetchChatReportsByIDs()` is looking for an `Array` and it calls `.join()` on it. So it's not clear how this code is working.

Reply:
For me `response.chatList` is always an Array containing Integers like so:
![image](https://user-images.githubusercontent.com/12156624/116244829-f7c52400-a770-11eb-8f85-ed280a2f8232.png)

Or an empty array like `[]` for new users

Are you saying there can be cases:
- a single number like `71402263` is returned
- a comma separated string is returned like `"71402263,71403817,71403849,71734617,71955477,72156283,72207410"`

In that case, yeah a special handling would be needed

Reply:
I may have misspoke, but I am remembering that it is "weird". 😂 

I think specifically it can be either an `Array` (like you are seeing) or a single number. Hence, why we were casting the array into a string before splitting it back into an array.

Reply:
Ok, in that case I'll revert the old logic, but add filtering on the resulting array so that the bug with the false positive is addressed. 

This is recent though as before, creating account would just fail due to the navigation issue, and then if you refresh everything is fine, while now if this is not addressed you will never enter the case where a Concierge conversation is created and you're stuck with no chats